src/gt4py/backend/dawn_backends.py

```diff
@@ -611,7 +611,7 @@ def make_extension_sources(self, gt_backend_t: str) -> Dict[str, Any]:
     def make_args_data(
         definition_ir: gt_ir.StencilDefinition, sir_field_info: Dict[str, Any]
     ) -> Dict[str, Any]:
-        data: Dict[str, Any] = {"field_info": {}, "parameter_info": {}, "unreferenced": {}}
+        data: Dict[str, Any] = {"field_info": {}, "parameter_info": {}, "unreferenced": []}
```

Comment:
I don't think it really matters, since the arguments are grabbed from the definition IR and this should guarantee no duplicates. But yes, `set` works just fine and feels better.